cloup/constraints/_core.py

```diff
@@ -513,12 +513,12 @@ def __init__(self, min: int, max: int):  # noqa
         :param min: must be an integer >= 0
         :param max: must be an integer > min
         """
+        super().__init__(RequireAtLeast(min) & AcceptAtMost(max), min=min, max=max)
         check_arg(min >= 0, 'min must be non-negative')
         if max is not None:
             check_arg(min < max, 'must be: min < max.')
```

Comment:
I think here and in the other the other constraint it makes sense to validate the arguments *before* calling `super()`, since you get better error messages and a better stack trace.

Reply:
Okay

Reply:
Let me know if it's any better now.

Reply:
Yep, perfect.